ci/wpf-dotnet-core.yml

```diff
@@ -0,0 +1,107 @@
+#
+# To configure this workflow:
+#
+# 1. Configure environment variables
+# GitHub sets default environment variables for every workflow run.  Replace the variables relative to your
+# project in the "env" section below.
+# 
+# 2. Signing
+# Generate a signing certificate in the Windows Application Packaging Project or add an existing signing certificate
+# to the project. Next, use PowerShell to encode the .pfx file using Base64 encoding by running the following Powershell
+# script to generate the output string.
+# 
+# $pfx_cert = Get-Content '.\SigningCertificate.pfx' -Encoding Byte
+# [System.Convert]::ToBase64String($pfx_cert) | Out-File 'SigningCertificate_Encoded.txt'
+#
+# Open the output file, SigningCertificate_Encoded.txt, and copy the string inside. Then, add the string to the repo
+# as a GitHub secret and name it "Base64_Encoded_Pfx."
+# For more information on how to configure your signing certificate for this workflow, refer
+# to https://github.com/microsoft/github-actions-for-desktop-apps#signing
+#
+# Finally, add the signing certificate password to the repo as a secret and name it "Pfx_Key."
+# See "Build the Windows Application Packaging project" below to see how the secret is used.
+#
+# For more information on GitHub Actions, refer to https://github.com/features/actions
+# For a complete CI/CD sample to get started with GitHub Action workflows for Desktop Applications,
+# refer to https://github.com/microsoft/github-actions-for-desktop-apps
+
+name:  WPF .NET Core
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+
+  build:
+
+    strategy:
+      matrix:
+        configuration: [Debug, Release]
+
+    runs-on: windows-latest  # For a list of available runner types, refer to 
+                             # https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on
+
+    env:
+      Solution_Name: your-solution-name                         # Replace with your solution name, i.e. MyWpfApp.sln.
+      Test_Project_Path: your-test-project-path                 # Replace with the path to your test project, i.e. MyWpfApp.Tests\MyWpfApp.Tests.csproj.
+      Wap_Project_Directory: your-wap-project-directory-name    # Replace with the Wap project directory relative to the solution, i.e. MyWpfApp.Package.
+      Wap_Project_Path: your-wap-project-path                   # Replace with the path to your Wap project, i.e. MyWpf.App.Package\MyWpfApp.Package.wapproj.
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    # Install the .NET Core workload
+    - name: Install .NET Core
+      uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: 3.1.101
+
+    # Add  MsBuild to the PATH: https://github.com/microsoft/setup-msbuild
+    - name: Setup MSBuild.exe
+      uses: microsoft/setup-msbuild@v1.0.0
```

Comment:
Can you specify the full 40 character SHA here instead of the specific version?  Since git tags are not immutable (but commit IDs are) we want to make sure that we're pinning to a particular version.

Here's the updated guidance for third-party actions in starter workflow templates:

> This workflow must only use actions that are produced by the language or ecosystem that the workflow supports. These actions must be published to the GitHub Marketplace. Workflows using these actions must reference the action using the full 40 character hash of the action's commit instead of a tag.

Reply:
@ethomson 
What would that look like?   I think I found the commit:
https://github.com/microsoft/setup-msbuild/commit/2008f912f56e61277eefaac6d1888b750582aa16

Like this?
   Add  MSBuild to the PATH: https://github.com/microsoft/setup-msbuild
    - name: Setup MSBuild.exe
    - uses: microsoft/setup-msbuild@2008f912f56e61277eefaac6d1888b750582aa16

Reply:
That looks like it's right based upon this guidance:
https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsuses